Busybox is built with /etc/shadow support by
default already. Dropbear was built without.

It seems this mismatch wasn't a problem for the
sdk's default business logic because it never
relied on this. Opensync does rely and therefore
make sure /etc/shadow is enabled in dropbear.

Ideally this should be a BUILD_USE_SHADOW_PASSWD
in the sdk which controls both busybox and
dropbear behavior.
--- a/userspace/public/apps/dropbear/Makefile
+++ b/userspace/public/apps/dropbear/Makefile
@@ -64,6 +64,7 @@ ALLOWED_LIB_DIRS := /lib:/lib/private:/l
 
 ifneq ($(strip $(BUILD_SSHD)),)
 
+SHADOW_FLAG = $(if $(strip $(BUILD_OPENSYNC)),--enable-shadow,--disable-shadow)
 check_untar_patch_configure: sanity_check
 	@if [ ! -e $(APP)/Makefile.in ]; then \
     	echo "Untarring original $(APP) source"; \
@@ -79,7 +80,7 @@ check_untar_patch_configure: sanity_chec
 	    fi; \
 	    (cd $(APP); ./configure CFLAGS="$(CFLAGS)" LDFLAGS="$(BCM_BLT32_FLAGS) -L$(INSTALL_DIR)/lib" LIBS="$(LIBS)" --host=$(TOOLCHAIN_PREFIX) \
         --target=$(TOOLCHAIN_PREFIX) \
-        --disable-largefile --disable-zlib  --disable-openpty --disable-syslog  --disable-shadow  --disable-lastlog  --disable-utmp --disable-utmpx --disable-wtmp --disable-wtmpx  \
+        --disable-largefile --disable-zlib  --disable-openpty --disable-syslog $(SHADOW_FLAG) --disable-lastlog  --disable-utmp --disable-utmpx --disable-wtmp --disable-wtmpx  \
         --prefix=/ \
         --sbindir=/bin); \
 	fi
